src/starfile/writer.py

```diff
@@ -118,75 +122,56 @@ def coerce_list(data_blocks: List[DataBlock]) -> Dict[str, DataBlock]:
     return {f'{idx}': df for idx, df in enumerate(data_blocks)}
 
 
-def write_blank_lines(file: Path, n: int):
-    with open(file, mode='a') as f:
-        f.write('\n' * n)
-
-
-def write_package_info(file: Path):
+def package_info():
     date = datetime.now().strftime('%d/%m/%Y')
     time = datetime.now().strftime('%H:%M:%S')
-    line = f'# Created by the starfile Python package (version {__version__}) at {time} on {date}'
-    with open(file, mode='w+') as f:
-        f.write(f'{line}\n')
+    return f'# Created by the starfile Python package (version {__version__}) at {time} on {date}'
 
 
-def write_simple_block(
-    file: Path,
+def simple_block(
     block_name: str,
     data: Dict[str, Union[str, int, float]],
     quote_character: str = '"',
     quote_all_strings: bool = False
-):  
+) -> Generator[str, None, None]:
+    yield f'data_{block_name}\n\n'
     quoted_data = {
-        k: f"{quote_character}{v}{quote_character}" 
-        if isinstance(v, str) and (quote_all_strings or " " in v or v == "") 
+        k: f"{quote_character}{v}{quote_character}"
+        if isinstance(v, str) and (quote_all_strings or " " in v or v == "")
         else v
-        for k, v
-        in data.items()    
+        for k, v in data.items()
     }
-    formatted_lines = '\n'.join(
-        [
-            f'_{k}\t\t\t{v}'
-            for k, v
-            in quoted_data.items()
-        ]
-    )
-    with open(file, mode='a') as f:
-        f.write(f'data_{block_name}\n\n')
-        f.write(formatted_lines)
-        f.write('\n\n\n')
+    for k, v in quoted_data.items():
+        yield f'_{k}\t\t\t{v}\n'
+    yield '\n' * 2
 
 
-def write_loop_block(
-    file: Path,
+def loop_block(
     block_name: str,
     df: pd.DataFrame,
     float_format: str = '%.6f',
     separator: str = '\t',
     na_rep: str = '<NA>',
     quote_character: str = '"',
     quote_all_strings: bool = False
-):
+) -> Generator[str, None, None]:
     # write header
     header_lines = [
         f'_{column_name} #{idx}'
         for idx, column_name
         in enumerate(df.columns, 1)
     ]
-    with open(file, mode='a') as f:
-        f.write(f'data_{block_name}\n\n')
-        f.write('loop_\n')
-        f.write('\n'.join(header_lines))
-        f.write('\n')
-
-    df = df.map(lambda x: f'{quote_character}{x}{quote_character}' 
-                if isinstance(x, str) and (quote_all_strings or " " in x or x == "") 
+    yield f'data_{block_name}\n\n'
+    yield 'loop_\n'
+    for line in header_lines:
+        yield line + '\n'
+
+    df = df.map(lambda x: f'{quote_character}{x}{quote_character}'
+                if isinstance(x, str) and (quote_all_strings or " " in x or x == "")
                 else x)
 
     # write data
-    df.to_csv(
-        path_or_buf=file,
+    yield df.to_csv(
```

Comment:
This does not yield the data part line by line, right?

Reply:
Indeed, that `yield` statement yields the entire CSV representation of the data. It might be nice to yield the individual lines:
```python3
for line in df.to_csv(...).split('\n'):
    yield line + '\n'
```